mqtt3/1.0/modules/ROOT/edits/mqtt3-connector-client-id.adoc

```diff
@@ -0,0 +1,355 @@
+= Set a Client ID with MQTT Connector
+
+Client IDs are important for MQTT because they identify a connection to an MQTT broker. Moreover, they identify a specific device or client,
+and it can be useful in terms of traceability to set a client ID that clearly identifies a device's connection to the broker.
+
+MQTT brokers usually implement a mechanism that closes an old connection if a new connection request with the same client ID as the existing (old) connection is received and then accepts the new connection. MQTT brokers employ this mechanism to prevent situations such as TCP connections reaching a half-open or invalid state due to a crash in one of the communication ends, which is undesirable. The broker assumes the new incoming connection to be valid, closes the older connection, and continues using the newly established connection.
+
+The client takeover mechanism can become a problem if you want to deploy an application in a cluster, and still provide an explicit client ID that identifies the client. Anypoint Connector for MQTT (MQTT Connector) provides client ID generators that help you set a static client ID, set a client ID with a dynamic component to adjust to different deployment scenarios, or define your custom client ID generator.
+
+== Configure the Client ID Custom Expression Generator
+
+This generator enables you to set a static string prefix and an expression as a suffix. Both parameters are optional but you must provide at least one of them in order to generate a client ID. If you do not provide any parameter, an `INVALID_CLIENT_ID` exception is thrown.
```

Comment:
So, the custom expression generator (or you?) creates a (custom?) client ID consisting of either or both of:

- A static string prefix (what does "static" mean, and for what body string is this the prefix?)
- A (custom?) expression as a suffix (do we need to clarify "string" vs. "expression"? To what body string is this the suffix?)

<opt prefix>body<opt suffix> (but you need at least one of prefix or suffix?)

Reply:
So, are we saying that the custom expression generator creates a (random) client ID that the user customizes by adding either or both of a prefix (prepended) and a suffix (appended), turning the ID into an expression?

Reply:
Maybe we should begin by calling the prefix and suffix "parameters" that act as a prefix and suffix to <what?>